modules/core/src/v2/coins/algoToken.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * @prettier
+ */
+import * as _ from 'lodash';
+import { BitGo } from '../../bitgo';
+import { Algo } from './algo';
+import { CoinConstructor } from '../coinFactory';
+import { BitGoJsError } from '../../errors';
+import { coins } from '@bitgo/statics';
+
+export interface AlgoTokenConfig {
+  name: string;
+  type: string;
+  coin: string;
+  network: string;
+  decimalPlaces: number;
+}
+
+export class AlgoToken extends Algo {
+  static readonly tokenPattern: RegExp = /[0-9]/;
+  public readonly tokenConfig: AlgoTokenConfig;
+  private readonly _code: string;
+  private readonly _issuer: string;
+
+  constructor(bitgo: BitGo, tokenConfig: AlgoTokenConfig) {
+    super(bitgo);
+    this.tokenConfig = tokenConfig;
+
+    const [tokenCoin, token] = _.split(this.tokenConfig.type, Algo.coinTokenPatternSeparator);
+    if (tokenCoin !== tokenConfig.coin) {
+      throw new BitGoJsError(`invalid coin found in token: ${this.tokenConfig.type}`);
+    }
+    if (!token || !token.match(AlgoToken.tokenPattern)) {
+      throw new BitGoJsError(`invalid token: ${this.tokenConfig.type}`);
+    }
+    [this._code, this._issuer] = _.split(token, '-');
+  }
+
+  static createTokenConstructor(config: AlgoTokenConfig): CoinConstructor {
+    return (bitgo: BitGo) => new AlgoToken(bitgo, config);
+  }
+
+  get type() {
+    return this.tokenConfig.type;
+  }
+
+  get name() {
+    return this.tokenConfig.name;
+  }
+
+  get coin() {
+    return this.tokenConfig.coin;
+  }
+
+  get network() {
+    return this.tokenConfig.network;
+  }
+
+  get code() {
+    return this._code;
+  }
+
+  get issuer() {
+    return this._issuer;
+  }
+
+  get decimalPlaces() {
+    return this.tokenConfig.decimalPlaces;
+  }
+
+  protected getAlgoNetwork() {
+    return this.tokenConfig.network === 'Testnet' ? coins.get('algo') : coins.get('talgo');
```

Comment:
this is going to return the `algo` or `talgo` coins, not the network. Are you sure that's what you want?

Reply:
I'm not sure if this is ok, because I don't know where will it is used, it's based on stellar.

Reply:
What does this method retiurn? Is it all the coin and tokens defined for algo/talgo?
It  will be appropriate to rename it as getAlgoTokens - unless there is a polymorphic code that calls getAlgoNetwork. However, since the name has `Algo` in it - this cannot be polymorphic. 
We can improve upon stellar implementation if there is opportunity.